### ISSUE TITLE
Update Cookie Consent for GDPR

### ISSUE THREAD

**Issue by Cleop** [dwyl/dwyl-site#458]
As a user 
I want to have the freedom to consent or decline from the use of Cookies
So that I know how my personal data is being used and that the EU law is being followed.

- [ ] Cookies are switched off by default
- [ ] A modal appears on every page which offers two options: 
    - Option 1 - Grant consent
    - Option 2 - Do not grant consent
- [ ] If a user clicks to consent the modal no longer appears.
- [ ] If a user who previously consented wishes to withdraw their consent they can click a link in the footer called 'Cookies' which takes them to a page where they can change their consent to none.
- [ ] When a user who gave consent then withdraws it all of the cookies already created for them is cleared.
- [ ] If a user clicks to decline consent the modal no longer appears (for a given time period when it will reappear again, say 30 days?)
- [ ] All cookies on the site check whether cookie consent has been granted before being set

**Questions:**
- Is it ok to use a cookie to record when a user has declined cookie consent? Ie. `cookieConsent = false`? @nelsonic do you have any thoughts on this?

**Comment by nelsonic** [comment]
@Cleop I feel that we don't _need_ cookies on the website except for for Google Analytics
and we can _easily_ "initialise" GA ***`after`*** cookie consent is given.
The way you have done this for "CI" is _roughly_ what we need for the dwyl site.
But we need a _detailed_ write-up of how it works (_think technical blog post_)
because this is highly re-useable content. 
Thanks! 👍

**Comment by Cleop** [comment]
@nelsonic I totally agree, dwyl site doesn't need many cookies which makes life easier! 

I was wondering more specifically whether you had any thoughts on if it's GDPR compliant to use a cookie to remember if someone chose actively _not_ to give cookie consent? As in, once you say you don't want cookies, can a cookie still be used if it's not storing identifiable personal data?

This is a question I have for both CI and dwyl site.

**Comment by nelsonic** [comment]
@Cleop nah, kinda by definition it's not kosher to use a cookie to store the rejection of cookies ...
For CI we should put the cookie preference in the Elm Model so that the person will not be "nagged" for the remainder of their _current_ session.
cookie prefs question/banner should be delayed e.g:
`onLoad setTimeout(function () { checkGDPRCookieConsent() }, 3000 )`

If people are going to _reject_ use of Cookies they are basically not going to get a great experience of using the CI app, but we shouldn't "nag" people on every page when they are just browsing ...

**Comment by Cleop** [comment]
Thanks @nelsonic, interesting.

In the case of CI, we have multiple elm apps running and therefore if a user were to navigate to another page which is running a different elm app then the information regarding their preferences would be lost. 

Can you think of a way to overcome this?

Or do you think we should pursue a different route altogether with this in mind?

**Comment by nelsonic** [comment]
@Cleop in the case of having _multiple_ Elm apps running on the _same_ website,
we need to get _creative_ ... 
First question **`before`** we dive into trying to solve a ***complex*** problem is: 
do we _need_ to have ***multiple*** Elm apps? (_or can we initialise the **same** app with different "flags" depending on the page being loaded?_)

**Comment by Cleop** [comment]
@nelsonic I like the idea of that approach, I think in these specific circumstances the time/budget constraint of the client and the scale of the app would make that unfeasible.

Can you think of any faster options or do you think that is realistically the fastest solution?

**Comment by Cleop** [comment]
Awesome, thanks for the insight @nelsonic, I've been able to implement this 👍 😊
